### PR TITLE
implement api auditEvents get by startAt and endAt timestamp with validations

### PR DESCRIPTION
**Pull request by shriramshankar**

### CODE REVIEW

File: api/v1/controllers/auditEvents.js [hunk 1]
```diff
@@ -14,10 +14,48 @@
...
 const u = require('../helpers/verbs/utils');
-const httpStatus = require('../constants').httpStatus;
+const constants = require('../constants');
+const httpStatus = constants.httpStatus;
 const doFind = require('../helpers/verbs/doFind');
 const doGet = require('../helpers/verbs/doGet');
+const apiErrors = require('../apiErrors');
+
+/**
+ * Function to validate the query parameters in the request
```
Comment (iamigo) [inline api/v1/controllers/auditEvents.js]: function-level comments do not need to start with "Function to..."

should just be `Validate the query parameters in teh request.`

File: api/v1/helpers/nouns/auditEvents.js [hunk 2]
```diff
@@ -9,10 +9,74 @@
...
+
+  // converts 15 days to ms and substracts it from the current date
+  const dateTime = Date.now() - time * timeUnitsInMS[timeUnit];
+  return new Date(dateTime);
+} // getDateTimeFromRelativeTime
+
+/**
+ * Function to modify the where clause passed to the database model. The
+ * startAt and endAt keys are replaced with the loggedAt key and assigned to the
+ * corresponding comparision operator
```
Comment (iamigo) [inline api/v1/helpers/nouns/auditEvents.js]: comparison

File: api/v1/controllers/auditEvents.js [hunk 3]
```diff
@@ -14,10 +14,48 @@
...
+ * @param  {Object} params  - The request params
+ */
+function validateQueryParams(params) {
```
Comment (iamigo) [inline api/v1/controllers/auditEvents.js]: are we validating that endAt >= startAt?

File: api/v1/helpers/verbs/doFind.js [hunk 4]
```diff
@@ -128,7 +128,8 @@ module.exports = function doFind(req, res, next, props) {
     redisCache.get(props.cacheKey, (cacheErr, reply) => {
       if (cacheErr || !reply) {
         // if err or no reply, get resuls from db and set redis cache
-        doFindResponse({ req, res, next }, props, opts, props.cacheKey, props.cacheExpiry);
+        doFindResponse({ req, res, next }, props, opts, props.cacheKey,
```
Comment (iamigo) [inline api/v1/helpers/verbs/doFind.js]: do we have to pass props.cacheKey and props.cacheExpiry separately given that the function is already getting the whole props object?

Comment (shriramshankar) [reply api/v1/helpers/verbs/doFind.js]: I just formatted this line as it was going over 80 characters.

File: api/v1/controllers/auditEvents.js (hunk 3 above)
Comment (shriramshankar) [reply api/v1/controllers/auditEvents.js]: Missed this validation. Will do it.

File: tests/api/v1/auditEvents/get.js [hunk 5]
```diff
@@ -182,4 +181,274 @@ describe('tests/api/v1/auditEvents/get.js >', () => {
...
+    it('Ok, relativeDateTime filter with seconds', (done) => {
+      const d = new Date();
+
+      // set date to current date + 3 seconds
+      d.setSeconds(d.getSeconds() + 3);
+      const ae = u.createAuditEventObject('Collector', 'GeneratorTemplate',
+        d, 'true');
+      AuditEvent.create(ae)
+      .then(() => {
+        api.get(`${path}?relativeDateTime=-s`)
```
Comment (pallavi2209) [inline tests/api/v1/auditEvents/get.js]: relativeDateTime=-s Is this valid? If not, do we need validation that there should be a number?

Comment (shriramshankar) [reply tests/api/v1/auditEvents/get.js]: relativeDateTime=-s or -h or -d or -m are converted to relativeDateTime=-0s .. Let me know if we want to validate it or not.

**Comment by shriramshankar** [comment]
@iamigo @pallavi2209 Updated the PR. let me know if we need the validation discussed above.

**Pull request merged**